### DISCUSSION TITLE
How best to refresh ALL elements/reload a page?

### DISCUSSION THREAD

**Discussion by groucho86** [zauberzeug/nicegui#1866] [Q&A]
### Question

I have a "reset" button in order for all user fields to revert to initial values. I was hoping there would be a reload function for an entire page but couldn't find one.

I'd rather avoid wrapping the entire code with a `@ui.refreshable` function, but maybe that's the cleanest solution?

I tried
```
def reset():
    ui.run_javascript('location.reload();')
```
but get
```
RuntimeWarning: coroutine 'run_javascript' was never awaited
  ui.run_javascript('location.reload();')
```

And if I do
```
async def reset():
    await ui.run_javascript('location.reload();')
```
It successfully reloads the page but I get:
```
    await ui.run_javascript('location.reload();')
  File "/opt/homebrew/lib/python3.11/site-packages/nicegui/functions/javascript.py", line 27, in run_javascript
    return await client.run_javascript(code, respond=respond, timeout=timeout, check_interval=check_interval)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/nicegui/client.py", line 145, in run_javascript
    raise TimeoutError('JavaScript did not respond in time')
TimeoutError: JavaScript did not respond in time
```

**Comment by rodja** [comment]
You could upgrade to NiceGUI 1.4, where `ui.run_javascript('location.reload();')` works without await. In 1.3 you'll need to write `await ui.run_javascript('location.reload();', response=False)`.

**Reply by jermatic** [reply]
is this the best way to make a page refreshable with nicegui?

my use case is a local dev gui where the dev can select envs like local/dev/stg/prd but the entire page needs to refresh in response to a selection change

**Reply by falkoschindler** [reply]
In [1.4.18](https://github.com/zauberzeug/nicegui/releases/tag/v1.4.18) we introduced `ui.navigate.reload()` to reload the page.
